org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/decorators/implicits/ImplicitHighlightingPresenter.scala

```diff
@@ -55,13 +55,12 @@ object ImplicitHighlightingPresenter {
       val txt = new String(sourceFile.content, t.pos.start, math.max(0, t.pos.end - t.pos.start)).trim()
       val pos = mkPosition(t.pos, txt)
       val region = new Region(pos.offset, pos.getLength)
-      val msg = compiler.asyncExec{
+      val msg = compiler.asyncExec {
         val sname = t.fun.symbol.nameString
         s"Implicit conversion found: `$txt`$DisplayStringSeparator`$sname($txt): ${t.tpe}`"
       }.getOption()
 
-      val annotation = new ImplicitConversionAnnotation(
-          () => compiler.mkHyperlink(t.symbol, name = "Open Implicit", region, scu.scalaProject.javaProject), msg.getOrElse(""))
+      val annotation = new ImplicitConversionAnnotation(t.pos, region, msg.getOrElse(""))
```

Comment:
Yes, the main problem is that the symbol refers to many other objects (it's nested, and has `outer` and `owner` fields), which really may keep a full compiler in memory. The ticket has some screenshots with memory heap dumps.